compiler/ir/ir.tree/tree-generator/src/org/jetbrains/kotlin/ir/generator/print/Elements.kt

```diff
@@ -71,7 +71,7 @@ fun printElements(generationPath: File, model: Model) = sequence {
                 }.build())
             }
 
-            if (element.visitorParent != null) {
+            if (element.visitorParent != null && (element.isLeaf || element.accept)) {
```

Comment:
Very minor: this logic (&&) would be best on the model layer, i.e. in the Element class, where it is copied from config